libs/designsystem/src/lib/components/checkbox/checkbox.component.ts

```diff
@@ -2,50 +2,50 @@ import {
   ChangeDetectionStrategy,
   Component,
   EventEmitter,
+  HostBinding,
   Input,
-  OnChanges,
   Output,
-  SimpleChange,
-  SimpleChanges,
 } from '@angular/core';
 
+// Counter for generating unique element ids
+let uniqueId = 0;
+
 @Component({
   selector: 'kirby-checkbox',
   templateUrl: './checkbox.component.html',
   styleUrls: ['./checkbox.component.scss'],
   changeDetection: ChangeDetectionStrategy.OnPush,
 })
-export class CheckboxComponent implements OnChanges {
-  @Input() checked: boolean;
-  @Input() disabled: boolean;
-  @Input() color: string = 'primary';
-  @Input() shape: string = 'square';
-  @Output() checkedChange = new EventEmitter<boolean>();
+export class CheckboxComponent {
+  @Input() checked: boolean = false;
+  @Input() attentionLevel: '1' | '2' = '2';
+
+  @HostBinding('class.has-label')
+  @Input()
+  labelText: string;
 
-  classes: string[] = [];
+  @HostBinding('class.error')
+  @Input()
+  hasError: boolean = false;
 
-  private readonly SHAPE_INDEX = 0;
-  private readonly COLOR_INDEX = 1;
+  @HostBinding('class.disabled')
+  @Input()
+  disabled: boolean = false;
+
```

Comment:
Consider disabling through a `disabled` attribute as this would be semantically better
```suggestion
  @Input()
  disabled = false;

  @HostBinding('attr.disabled')
  get _isDisabled() {
    return this.disabled ? 'disabled' : null;
  }
```